config/recipes/gclb/01-elastic-stack.yaml

```diff
@@ -0,0 +1,74 @@
+---
+apiVersion: elasticsearch.k8s.elastic.co/v1
+kind: Elasticsearch
+metadata:
+  name: hulk
+  labels:
+    app: hulk
+spec:
+  version: 7.6.1
+  http:
+    service:
+      metadata:
+        labels:
+          app: hulk
+        annotations:
+          # Enable TLS between GCLB and the application
+          cloud.google.com/app-protocols: '{"https":"HTTPS"}'
+          service.alpha.kubernetes.io/app-protocols: '{"https":"HTTPS"}'
+          # Uncomment the following line to enable container-native load balancing.
+          #cloud.google.com/neg: '{"ingress": true}'
+      spec:
+        type: NodePort
```

Comment:
Does this need to be NodePort? It's not clear reading the GKE docs but the examples show ClusterIP. My gut feeling is that you pretty much always want to use ClusterIP unless there's a good reason to not.

Reply:
If you are not using container-native load balancing, the service has to be a `NodePort` because the load balancer sends traffic to the node which then gets forwarded by the kube proxy to the container.

Reply:
👍 with a comment to that effect I'm good to merge this in. I think in general I lean towards having the "recommended" way in the recipes also be the GKE recommended way (i.e. vpc native clusters), but as long as we document how to use them for VPC native clusters that should be enough to get people going

Reply:
The Google example uses a NodePort: https://cloud.google.com/kubernetes-engine/docs/how-to/container-native-load-balancing#create_service

But, I agree with you that there's no need to make the service do more than it really needs to.